src/pil/compiler/powdr_pil.rs

```diff
@@ -0,0 +1,184 @@
+use crate::{
```

Comment:
@qwang98 the compiler is not specific for the POWDR PIL, it will be the same for the polygon PIL, so all this code should be moved to mod.rs

Reply:
I agree. The real difference comes from how we convert IR to backend PIL string. I think there could be a chance someone writes a Polygon PIL converter from our PIL IR to Polygon PIL string.